src/models/linear_network.py

```diff
@@ -167,20 +176,35 @@ def _get_background(self, sample_size: int = 100) -> torch.Tensor:
                                       shuffle_data=True, mode='train',
                                       to_tensor=True)
         output_tensors: List[torch.Tensor] = []
+        if self.include_pred_month:
+            output_pred_months: List[torch.Tensor] = []
         samples_per_instance = max(1, sample_size // len(train_dataloader))
 
         for x, _ in train_dataloader:
             while len(output_tensors) < sample_size:
                 for _ in range(samples_per_instance):
-                    idx = random.randint(0, x.shape[0] - 1)
-                    output_tensors.append(x[idx])
-        return torch.stack(output_tensors)
+                    idx = random.randint(0, x[0].shape[0] - 1)
+                    output_tensors.append(x[0][idx])
+                    if self.include_pred_month:
+                        output_pred_months.append(self._one_hot_months(x[1][idx: idx + 1]))
+        if self.include_pred_month:
+            return [torch.stack(output_tensors), torch.cat(output_pred_months, dim=0)]
+        else:
+            return torch.stack(output_tensors)
+
+    @staticmethod
+    def _one_hot_months(indices: torch.Tensor) -> torch.Tensor:
+        return torch.eye(14)[indices.long()][:, 1:-1]
```

Comment:
what is the identity matrix doing here? is the output a `14 x 12 matrix`?

Reply:
The output is a `indices.shape[0] * 12` matrix.

You take the identity matrix, and take the indices in the `indices` array; this is the one hot encoding.

We ignore the 0th (since there is no 0 month) and the extra class.